### PR TITLE
Add support for XBPS package manager

### PR DESCRIPTION
**Pull request by SabariVig**
xbps package manager is used with void linux https://voidlinux.org/

### CODE REVIEW

File: lib/src/actions/package/providers/xbps.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        };
+        match which("xbps-query") {
+            Ok(_) => {
+                query = true;
+            }
+            Err(_) => {
+                warn!(message = "xbps-query not available");
+            }
+        };
+        install && query
```
Comment (rawkode) [inline lib/src/actions/package/providers/xbps.rs:39]: Can the query command be installed in bootstrap if missing?

Comment (SabariVig) [reply lib/src/actions/package/providers/xbps.rs:39]: No, void ships with both these binaries as inside single package called  xbps, so if one is installed other would also be installed

**Comment by rawkode** [comment]
@SabariVig I don’t have the means to test this, so if you say it works I’m happy to merge. 

I really need to work out some container based testing 😀

**Comment by SabariVig** [comment]
Feel free to merge it, ran few basic installs "it works on my machine" :)

**Comment by rawkode** [comment]
Thanks for contributing

**Pull request merged**